### PR TITLE
Add a main entry point to the Python package

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

The entry point currently only supports:

```
$ spectre --version
```

I intend to add some basic CLI functionality that calls into our Python package, e.g. to clean output, validate an input file, schedule runs, etc.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: cmake/SpectreSetupPythonPackage.cmake [hunk 1]
```diff
@@ -47,10 +60,16 @@ configure_file(
...
+# Install the SpECTRE Python package to the CMAKE_INSTALL_PREFIX, using pip.
+# This will install the package into the expected subdirectory, typically
+# `lib/pythonX.Y/site-packages/`. It also creates symlinks to entry points
+# specified in `setup.py`.
 install(
-  DIRECTORY ${SPECTRE_PYTHON_PREFIX}
-  DESTINATION ${SPECTRE_PYTHON_INSTALL_LIBDIR}
+  CODE "execute_process(\
+    COMMAND ${Python_EXECUTABLE} -m pip install \
+      --no-deps --no-input --no-cache-dir --no-index --ignore-installed \
```
Comment (nikwit) [inline cmake/SpectreSetupPythonPackage.cmake:70]: why no-cache-dir?

**Review by nikwit — DISMISSED** [review]
LGTM

File: cmake/SpectreSetupPythonPackage.cmake (hunk 1 above)
Comment (nilsvu) [reply cmake/SpectreSetupPythonPackage.cmake:70]: No need to cache anything here I think 🤷‍♂️

**Comment by nilsvu** [comment]
Switched to [setuptool's recommended setup.cfg](https://setuptools.pypa.io/en/latest/userguide/quickstart.html) after discussion with @nikwit, and added a basic CI test.

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**